sql/core/src/main/scala/org/apache/spark/sql/sources/commands.scala

```diff
@@ -417,15 +428,14 @@ private[sql] class DefaultWriterContainer(
       assert(writer != null, "OutputWriter instance should have been initialized")
       writer.close()
       super.commitTask()
-    } catch {
-      case cause: Throwable =>
-        super.abortTask()
-        throw new RuntimeException("Failed to commit task", cause)
+    } catch { case cause: Throwable =>
+      throw new RuntimeException("Failed to commit task", cause)
```

Comment:
Actually, I think we need to also add doc to `InsertIntoHadoopFsRelation` to explain the flow of this command and how we handle different kinds of failures/errors.